### PR TITLE
[vcpkg baseline][h5py-lzf] Update, use liblzf port, fix export, cleanup

### PR DESCRIPTION
**Pull request by dg0yt**
Stop using vendored liblzf in h5py-lzf, fixing `lzf.h` install conflict.
Alternative to https://github.com/microsoft/vcpkg/pull/37240.

### CODE REVIEW

File: ports/h5py-lzf/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+find_package(unofficial-liblzf CONFIG REQUIRED)
 
-add_library(h5py-lzf STATIC lzf_filter.c lzf/lzf_c.c lzf/lzf_d.c)
+add_library(h5py-lzf STATIC lzf_filter.c)
 target_include_directories(h5py-lzf PUBLIC $<INSTALL_INTERFACE:include>)
-target_link_libraries(h5py-lzf PRIVATE libzstd $<IF:${link_hdf5_SHARED},hdf5::hdf5-shared,hdf5::hdf5-static>)
-install(TARGETS h5py-lzf EXPORT h5py-lzf-config ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
-install(EXPORT h5py-lzf-config DESTINATION share/cmake/h5py-lzf)
-install(FILES lzf_filter.h lzf/lzf.h DESTINATION include)
+target_link_libraries(h5py-lzf PRIVATE unofficial-liblzf-liblzf $<IF:$<TARGET_EXISTS:hdf5::hdf5-shared>,hdf5::hdf5-shared,hdf5::hdf5-static>)
```
Comment (WangWeiLin-MV) [inline ports/h5py-lzf/CMakeLists.txt:9]:
```suggestion
target_link_libraries(h5py-lzf PRIVATE unofficial::liblzf::liblzf $<IF:$<TARGET_EXISTS:hdf5::hdf5-shared>,hdf5::hdf5-shared,hdf5::hdf5-static>)
```

**Review by WangWeiLin-MV — CHANGES_REQUESTED** [review]

File: ports/h5py-lzf/CMakeLists.txt (hunk 1 above)
Comment (dg0yt) [reply ports/h5py-lzf/CMakeLists.txt:9]: Oops :facepalm:

**Review by WangWeiLin-MV — APPROVED** [review]
Port usage tests pass with following triplets:
* x64-windows
* x64-windows-static

**Comment by BillyONeal** [comment]
Thanks for fixing our baseline issue!

**Pull request merged**